### PR TITLE
Feat/support min max for dates

### LINKED ISSUE #25: [Date] Add support for min and max
This should behave similar to numeric min/max

Example config:
```
date: {min: "2002-02-02", max: "2016-10-16"}
```

It should return `date.min` or `date.max` when it fails.

### PR DESCRIPTION
**Pull request by PeterCervienka**
Supporting min and max limits for dates.

Fixes #25

### CODE REVIEW

File: src/rules/__tests__/date_tests.js [hunk 1]
```diff
@@ -35,3 +35,18 @@ test('rules.date should return "date.future" when date is not in future', t => {
   t.is(date('field', inPast, { future: true }), 'date.future');
   t.is(date('field', moment(), { future: true }), 'date.future');
 });
+
```
Comment (relekang) [inline src/rules/__tests__/date_tests.js]: The min/max config values should be strings.

**Comment by relekang** [comment]
👌 

Can you fix the breaking test?

File: src/rules/__tests__/date_tests.js (hunk 1 above)
Comment (PeterCervienka) [inline src/rules/__tests__/date_tests.js]: changed to strings in format YYYY-MM-DD

**Comment by PeterCervienka** [comment]
failing tests fixed

**Pull request merged**